dandi/tests/test_metadata.py

```diff
@@ -1265,7 +1302,421 @@ def test_dandimeta_migration(schema_dir):
+            dict(
+                name="Neuroscience, McKnight Endowment for",
+                roleName=["dandi:Sponsor"],
+                includeInCitation=True,
+            ),
+            dict(
+                name=(
+                    "Foundation, NARSAD Young Investigator grant from the Brain"
+                    " & Behavior Research"
+                ),
+                roleName=["dandi:Sponsor"],
+                includeInCitation=True,
+            ),
+            dict(
+                name="Foundation, Kavli",
+                roleName=["dandi:Sponsor"],
+                includeInCitation=True,
+            ),
+            dict(
+                awardNumber="U19NS104590",
+                name="initiative, BRAIN",
+                roleName=["dandi:Sponsor"],
+                includeInCitation=False,
+            ),
+        ],
+        description=(
+            "A challenge for data sharing in systems neuroscience is the"
+            " multitude of different data formats used. Neurodata Without"
+            " Borders: Neurophysiology 2.0 (NWB:N) has emerged as a"
+            " standardized data format for the storage of cellular-level data"
+            " together with meta-data, stimulus information, and behavior. A"
+            " key next step to facilitate NWB:N adoption is to provide easy to"
+            " use processing pipelines to import/export data from/to NWB:N."
+            " Here, we present a NWB-formatted dataset of 1863 single neurons"
+            " recorded from the medial temporal lobes of 59 human subjects"
+            " undergoing intracranial monitoring while they performed a"
+            " recognition memory task. We provide code to analyze and"
+            " export/import stimuli, behavior, and electrophysiological"
+            " recordings to/from NWB in both MATLAB and Python. The data files"
+            " are NWB:N compliant, which affords interoperability between"
+            " programming languages and operating systems. This combined data"
+            " and code release is a case study for how to utilize NWB:N for"
+            " human single-neuron recordings and enables easy re-use of this"
+            " hard-to-obtain data for both teaching and research on the"
+            " mechanisms of human memory."
+        ),
+        identifier=dict(value="000004", propertyID="DANDI"),
+        keywords=[
+            "cognitive neuroscience",
+            "data standardization",
+            "decision making",
+            "declarative memory",
+            "neurophysiology",
+            "neurosurgery",
+            "NWB",
+            "open source",
+            "single-neurons",
+        ],
+        license=["dandi:CCBY40"],
+        name=(
+            "A NWB-based dataset and processing pipeline of human single-neuron"
+            " activity during a declarative memory task"
+        ),
+        schemaVersion="1.0.0-rc1",
+        repository="https://dandiarchive.org/",
+    )
```

Comment:
just a note: although I love tests, I think eventually such "sample test" (instead of just concise targetted unittest which would test on some smaller metadata record) might become too much of a burden to maintain